Amazon.Ion.ObjectMapper/IonSerializer.cs

```diff
@@ -131,16 +135,54 @@ public interface IonSerializerFactory<T, TContext> where TContext : IonSerializa
 
     public class IonSerializer
     {
-        private readonly IonSerializationOptions options;
+        internal readonly IonSerializationOptions options;
+        private Dictionary<Type, dynamic> primitiveSerializers { get; init; }
+        private IonNullSerializer nullSerializer { get; init; }
+        private IonClobSerializer clobSerializer { get; init; }
+        private IonListSerializer listSerializer { get; init; }
+        private IonObjectSerializer objectSerializer { get; init; }
 
         public IonSerializer() : this(new IonSerializationOptions())
         {
-
         }
 
         public IonSerializer(IonSerializationOptions options)
         {
             this.options = options;
+            this.primitiveSerializers = new Dictionary<Type, dynamic>()
+            {
+                {typeof(bool), new IonBooleanSerializer()},
+                {typeof(string), new IonStringSerializer()},
+                {typeof(byte[]), new IonByteArraySerializer()},
+                {typeof(int), new IonIntSerializer()},
+                {typeof(long), new IonLongSerializer()},
+                {typeof(float), new IonFloatSerializer()},
+                {typeof(double), new IonDoubleSerializer()},
+                {typeof(decimal), new IonDecimalSerializer()},
+                {typeof(BigDecimal), new IonBigDecimalSerializer()},
+                {typeof(SymbolToken), new IonSymbolSerializer()},
+                {typeof(DateTime), new IonDateTimeSerializer()},
+                {typeof(Guid), new IonGuidSerializer(this.options.AnnotateGuids)},
+            };
+            this.nullSerializer = new IonNullSerializer();

```

Comment:
I don't know why these are now fields.

Reply:
No longer using these fields for these serializers. Instead we just spawn new ones each time we need one.